### ISSUE TITLE
pvRequest - asking for subfield of non structure field crashes IOC; allow subfield of union

### ISSUE THREAD

**Issue by mrkraimer** [epics-base/pvDatabaseCPP#41]
The following causes the IOC to crash:

```
mrk> pvinfo PVRdouble
PVRdouble
Server: 10.0.0.19:5075
Type:
    epics:nt/NTScalar:1.0
        double value
        alarm_t alarm
            int severity
            int status
            string message
        time_t timeStamp
            long secondsPastEpoch
            int nanoseconds
            int userTag

mrk> pvget -r value.x PVRdouble
```

There was also a request from Guobao Shen to allow a request a subfield of a restricted union
field.

It would succeed if the subfield is the type for the current array field and throw an exception
otherwise.

For example

```
mrk> pvinfo PVRrestrictedUnion
PVRrestrictedUnion
Server: 10.0.0.19:5075
Type:
    epics:nt/NTUnion:1.0
        union value
            string string
            string[] stringArray
        time_t timeStamp
            long secondsPastEpoch
            int nanoseconds
            int userTag

mrk> pvget -r value.string PVRrestrictedUnion

```
This would succeed if the current value has type string and throw an exception otherwise.

**Comment by mrkraimer** [comment]
This is fixed by pull request #42.

**Comment by mrkraimer** [comment]
pull request #42 address this issue
